### PR TITLE
feat: new dependency graph

### PR DESCRIPTION
**Pull request by mateuszjenek**
Replace old graph library with the new Humanitec Resource Graph component

Result:
Zrzut ekranu 2024-12-2 o 14 43 06

### CODE REVIEW

**Comment by mathieu-benoit** [comment]
@mateuszjenek, anything missing here to merge this and publish a new version?

File: src/adapters/humctl/HumctlAdapter.ts [hunk 1]
```diff
@@ -88,11 +88,20 @@ export class HumctlAdapter implements IHumctlAdapter {
       result.stderr.includes('HTTP-403')
     ) {
       throw new UnauthorizedError();
-    } else if (result.stderr.includes('environment is required')) {
+    } else if (
+      result.stderr.includes('environment is required') ||
+      result.stderr.includes('missing env')
```
Comment (damien-duignan) [inline src/adapters/humctl/HumctlAdapter.ts:93]: Where are these new errors coming from?

File: src/repos/DeploymentRepository.ts [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+  ): Promise<unknown> {
+    const deploymentUrl =
+      '/orgs/' +
+      organizationId +
+      '/apps/' +
+      applicationId +
+      '/envs/' +
+      environmentId +
+      '/deploys/' +
+      deployId;
```
Comment (damien-duignan) [inline src/repos/DeploymentRepository.ts:30]:
```suggestion
    const deploymentUrl = `/orgs/${organizationId}/apps/${applicationId}/envs/${environmentId}/deploys/${deployId}`;
```

**Review by damien-duignan — CHANGES_REQUESTED** [review]

File: src/adapters/humctl/HumctlAdapter.ts (hunk 1 above)
Comment (mateuszjenek) [reply src/adapters/humctl/HumctlAdapter.ts:93]: CLI returns unconsistent error messages and I found out that in most places it returns: `environment is required`, but in some: `missing env`

**Review by damien-duignan — APPROVED** [review]

**Pull request merged**